src/_includes/docs/swift-package-manager/migrate-objective-c-plugin.md

```diff
@@ -0,0 +1,365 @@
+    -  objcHeaderOut: 'ios/Classes/messages.g.h',
+    -  objcSourceOut: 'ios/Classes/messages.g.m',
+    +  objcHeaderOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/messages.g.h',
+    +  objcSourceOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/messages.g.m',
+       copyrightHeader: 'pigeons/copyright.txt',
+    ```
+
+    If your `objcHeaderOut` file is no longer within the same directory as the
+    `objcSourceOut`, you can change the `#import` using
+    `ObjcOptions.headerIncludePath`:
+
+    ```diff2html
+    --- a/pigeons/messages.dart
+    +++ b/pigeons/messages.dart
+    @@ -18,8 +18,8 @@ import 'package:pigeon/pigeon.dart';
+       javaOptions: JavaOptions(),
+    -  objcHeaderOut: 'ios/Classes/messages.g.h',
+    -  objcSourceOut: 'ios/Classes/messages.g.m',
+    +  objcHeaderOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/include/plugin_name_ios/messages.g.h',
+    +  objcSourceOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/messages.g.m',
+    +  objcOptions: ObjcOptions(
+    +    headerIncludePath: './include/plugin_name_ios/messages.g.h',
+    +  ),
+       copyrightHeader: 'pigeons/copyright.txt',
+    ```
+
+   Run Pigeon to re-generate its code with the latest configuration.
+
+13. Update your `Package.swift` file with any customizations you might need.
+
+    1. Open `/plugin_name/plugin_name_ios/ios/plugin_name_ios/` in Xcode.
+
+        * If package does not show any files in Xcode, quit Xcode (**Xcode >
+          Quit Xcode**) and reopen.
+
+        * You don't need to edit your `Package.swift` file through Xcode,
+          but Xcode provides helpful feedback.
+
+        * If Xcode isn't updating after you make a change, try clicking
+          **File > Packages > Reset Package Caches**.
+
+    1. [Add dependencies][].
+
+    2. If your package must be linked explicitly `static` or `dynamic`
+       ([not recommended by Apple][]), update the [Product][] to define the
+       type:
+
+       ```swift title="Package.swift"
+       products: [
+           .library(name: "plugin-name-ios", type: .static, targets: ["plugin_name_ios"])
+       ],
+       ```
+
+    3. Make any other customizations. For more information on how to write a
+       `Package.swift` file, see
+       [https://developer.apple.com/documentation/packagedescription](https://developer.apple.com/documentation/packagedescription).
+
+       :::tip
+       If you add additional targets to your `Package.swift` file,
+       try to name them uniquely.
+       If your target name conflicts with another target from another package,
+       this can cause issues for developers that use your plugin.
+       :::
+
+1.  Update your `plugin_name_ios.podspec` to point to new paths.
+
+    ```diff2html
+    --- a/plugin_name_ios.podspec
+    +++ b/plugin_name_ios.podspec
+    @@ -21,4 +21,4 @@ 
+    - s.source_files = 'Classes/**/*.{h,m}'
+    - s.public_header_files = 'Classes/**/*.h'
+    - s.module_map = 'Classes/cocoapods_plugin_name_ios.modulemap'
```

Comment:
This `podspec` contains attributes that customers might not have.

For example, [only ~70 out of ~10K plugins have a `modulemap`](https://gist.github.com/loic-sharma/891a358f8abab36133e8a9395f719296). Folks that don't have a `modulemap` don't need to add/change a `module_map` attribute.

Hopefully that's clear, but let me know if you have recommendations to improve clarity here.